tests/test_api.py

```diff
@@ -187,3 +188,48 @@ def test_ask_analysis_result(self):
         data = {"job_id": job_id}
         response = self.client.get("/api/ask_analysis_result", data)
         self.assertEqual(response.status_code, 200)
+
+    def test_list_all_jobs(self):
+        response = self.client.get("/api/jobs")
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_job_by_id(self):
+        job_id = os.environ.get("TEST_JOB_ID","1")
+        response = self.client.get(f"/api/jobs/{job_id}")
+        self.assertEqual(response.status_code, 200)
+
+class ApiTagTests(ApiTests):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = User.objects.create_user(
+            username="test", email="test@intelowl.com", password="test"
+        )
+        self.client.force_authenticate(user=self.user)
+        self.tag = models.Tag.objects.create(label="Test", color="#FF5733")
+
+    def test_create_new_tag(self):
+        data = {"label": self.tag.label, "color": self.tag.color}
```

Comment:
the `label` and `color` here should be different. See my message on slack group.

Reply:
```python3
self.assertEqual(models.Tag.objects.count(), 1)
data= {"label": "newTestLabel", "color": "#765A54"}
response = self.client.post("/api/tags", data)
self.assertEqual(response.status_code, 200)
self.assertEqual(models.Tag.objects.count(), 2)
```

Reply:
Listing and creation tests for both jobs and tags work but the rest still give a `404`. Also, still not sure how to go about getting sufficient permission for `delete`.

Reply:
there is no PUT/DELETE for jobs, only for tags

Reply:
Yes, sorry I meant listing works for jobs (GET) but get-by-id doesn't. For tags, create and list tags (GET) work but the rest don't, with delete giving a 403.

Reply:
This is because the test job you create in `setUp` is via admin user and the test user doesn't have permission to view/delete it, 

So we should do 2 test cases:
1. For the `job` created in `setUp`, test user doesn't have permission so the API gives 403.
2. In the `test_job_get_by_id` and `test_job_delete_by_id` functions, we can use the guardian's `assign_perm` method  [(ref.)](https://django-guardian.readthedocs.io/en/stable/api/guardian.shortcuts.html#assign-perm) to give user permission to the test `job` and then we should receive 200 status code.

Reply:
Yeah I was also thinking permissions were the issue since I tried sending a request manually via `curl` using `durin-auth- token` from my `django-admin-interface` and then it seemed to work.

So, I should split  `test_get<tag/job>_by_id`, `test_update_tag` and `test_delete_tag` tests each into 2 tests where in one i'm confirming that the user doesn't have enough permission to make those changes and in the other i'm giving the permission and then confirming that now they can? Did I understood that correctly?

Reply:
Yes. And you are right, get_by_id, put and delete.